### PR TITLE
API-8373 FIX KR ORDER CREATION

### PR DESCRIPTION
**Pull request by tom-unicity**
bypass fallback query when state has non english characters

### CODE REVIEW

**Review by wallter — APPROVED** [review]
co-authored & debugged

File: classes/Unicity/Locale/State.php [hunk 1]
```diff
@@ -163,22 +163,24 @@ protected function getStateData($state, $country)
...
-                    ->where(DB\SQL::expr("LOWER(PREG_REPLACE('/[^a-z]/i', '', [StateAlias]))"), '=', strtolower(preg_replace('/[^a-z]/i', '', $state_2)), 'OR')
-                    ->where_block(')')
-                    ->where('CountryNumeric3', '=', $country)
-                    ->limit(1)
-                    ->query();
-
-                if ($records->is_loaded()) {
-                    return $records->current();
-                }
+				if (preg_replace('/[^a-z]/i', '', $state_1) !== '') {
```
Comment (DerekHansenUnicity) [inline classes/Unicity/Locale/State.php:166]: NIT: The indentation seems wrong here

Tabs vs spaces?

**Review by DerekHansenUnicity — APPROVED** [review]

**Review by DerekHansenUnicity — CHANGES_REQUESTED** [review]
I'd like the indentation fixed before mergins

File: classes/Unicity/Locale/State.php (hunk 1 above)
Comment (tom-unicity) [reply classes/Unicity/Locale/State.php]: done.

**Review by DerekHansenUnicity — APPROVED** [review]

**Comment by DerekHansenUnicity** [comment]
btw the comments are very helpful

**Pull request merged**